generative_ai/streaming_chat.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START aiplatform_streaming_chat]
+import vertexai
+from vertexai.language_models import ChatModel, InputOutputTextPair
+
+
+def streaming_prediction(
+    project_id: str,
+    location: str,
+) -> str:
+    """Streaming Chat Example with a Large Language Model"""
+
+    vertexai.init(project=project_id, location=location)
+
+    chat_model = ChatModel.from_pretrained("chat-bison")
+
+    parameters = {
+        "temperature": 0.8,  # Temperature controls the degree of randomness in token selection.
+        "max_output_tokens": 256,  # Token limit determines the maximum amount of text output.
+        "top_p": 0.95,  # Tokens are selected from most probable to least until the sum of their probabilities equals the top_p value.
+        "top_k": 40,  # A top_k of 1 means the selected token is the most probable among all tokens.
+    }
+
+    chat = chat_model.start_chat(
+        context="My name is Miles. You are an astronomer, knowledgeable about the solar system.",
+        examples=[
+            InputOutputTextPair(
+                input_text="How many moons does Mars have?",
+                output_text="The planet Mars has two moons, Phobos and Deimos.",
+            ),
+        ],
+    )
+
+    responses = chat.send_message_streaming(
+        message="How many planets are there in the solar system?", **parameters
+    )
+
+    results = ""
+    for response in responses:
+        print(response)
+        results += str(response)
```

Comment:
Can this work instead? String addition can also result in quadratic runtime, so `''.join(list[str])` is much preferred.
```suggestion
    results = "\n".join(str(response) for response in responses)
    print(results)
```

Reply:
No, unfortunately this doesn't work. I tried multiple times. It might because it's streaming.

Reply:
SG - how about this then?
```
results = []
for response in responses:
        print(response)
        results.append(str(response))
results = "\n".join(results)
```

Reply:
Let me try!